File: performance-metrics-java/src/main/java/com/metrics/performancemetrics/entity/MetricValue.java
package com.metrics.performancemetrics.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.annotation.Nonnull;
import jakarta.persistence.*;
import lombok.Data;

import java.time.Instant;
import java.time.LocalDateTime;

@Entity(name = "metric_value")
@Data
public class MetricValue {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    @Column(nullable = false)
    private Double value;
    @Column(nullable = false)
    private Instant created_at;
    @ManyToOne
    @JoinColumn(name = "metric_id", nullable = false)
    @JsonIgnore
    private Metric metric;
}

File: performance-metrics-android/app/src/main/java/com/metrics/performancemetrics/data/MetricValue.kt
package com.metrics.performancemetrics.data

import com.google.gson.annotations.SerializedName
import com.metrics.performancemetrics.util.DateFormatter
import java.time.Instant
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter

data class MetricValue(@SerializedName("id") val id : Int,
                       @SerializedName("value") val value : Double,
                       @SerializedName("created_at") val createdAt : String,
                       @SerializedName("metric_id") val metricId : Int)
fun MetricValue.createdAtMillis() : Long
{
    return DateFormatter.formatStringDateToMillis(createdAt)
}